Add cc.arduino.IDE2

joshuag1000 opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** (https://github.com/arduino/arduino-ide/issues/1001)
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

This pr requests that version 2 of the arduino ide be added to flathub. v2 is an entire remake of the ide and after talking it through with the maintainers of the v1 version we agreed it would best to have the second version of the app to be added as a new app (https://github.com/flathub/cc.arduino.arduinoide/pull/42).

Continuation of #3176. 
I had to rename the app to fit what was already used by the developers in the code (https://github.com/arduino/arduino-ide/blob/0a87fd00f38c923a36ce22d9a787fd071a6173e7/electron/packager/config.js#L137). sorry for that.

joshuag1000 commented:
@[hfiguiere](https://github.com/hfiguiere) As for the dbus permission I had a look into the code for the ide and the dbus appears to be used in the arduino-cli part of the application by some libraries. And while its not ideal its probably best i leave this permission in until i can determine what it is fully used for in the cli and if it is safe to remove.
(see: https://github.com/arduino/arduino-cli) (library used: https://github.com/godbus/dbus)

joshuag1000 commented:
bot, build cc.arduino.IDE2

hfiguiere commented:
you did need to resubmit at all.

hfiguiere commented on cc.arduino.IDE2.json:
```diff
@@ -0,0 +1,116 @@
+    "separate-locales": false,
+    "finish-args": [
+        "--socket=x11",
+        "--socket=system-bus",
```
I'll put this one again since it was on the previous PR.

You need to figure which dbus names this want to access.

hfiguiere commented on flathub.json:
```diff
@@ -0,0 +1,4 @@
+{
+    "only-arches": ["x86_64"]
```
and for the record it is still disappointing that RaspberryPi users are left out of this. Programming Arduino and using a RespberryPi goes hand in hand.

joshuag1000 replied:
agreed. but there isn't much i can do about this except for wait for support for arm based cpu's.

hfiguiere replied:
building from source instead of pre-built binaries.

hfiguiere replied:
but I repeated that comment since you closed the previous PR.

joshuag1000 replied:
> building from source instead of pre-built binaries.

yeah i will consider this once i can get the application to compile for x86_64 first, but with support for arm still being experimental at best i dont believe its best to add it until the devs deem it almost ready. This is one of the reasons making a new app on flathub is a good idea as users may still want to use the old ide especially on arm.

joshuag1000 commented:
> you did NOT need to resubmit at all.
> 
> you could have updated the previous PR and kept all the comments.

I wasn't sure how the new repos were generated by the system so when I updated the name inside the JSON etc I updated the branch name which in turn auto closed the pr

hfiguiere commented:
just for the record, the branch name is totally irrelevant.

joshuag1000 commented:
> just for the record, the branch name is totally irrelevant.

ok i'll make a note of that for any future requests i might make.

joshuag1000 commented:
bot, build cc.arduino.IDE2

hfiguiere commented on cc.arduino.IDE2.json:
```diff
@@ -0,0 +1,117 @@
+        "--device=all",
+        "--filesystem=home",
+        "--filesystem=xdg-run/keyring",
+        "--system-talk-name=org.freedesktop.DBus",
```
I'm not sure this is the right value either.

joshuag1000 replied:
from what i can see the application talks with various dbus functions and adding each one individually seems like a lot of unneeded effort. i had a look through some of the other apps on flathub and they dont appear to do that.

joshuag1000 replied:
some examples include:
`C17: -> org.freedesktop.DBus call org.freedesktop.DBus.ListNames at /org/freedesktop/DBus
C18: -> org.freedesktop.DBus call org.freedesktop.DBus.RemoveMatch at /org/freedesktop/DBus
C16: -> org.freedesktop.DBus call org.freedesktop.DBus.AddMatch at /org/freedesktop/DBus
`

joshuag1000 replied:
ok so after monitoring my dbus i believe i've figured out what dbus names i need to add so i am going to resolve this and i have pushed a commit with the new dbus name.

alranel commented on cc.arduino.IDE2.appdata.xml:
```diff
@@ -0,0 +1,70 @@
+	<project_license>AGPL-3.0</project_license>
+	<name>Arduino IDE v2</name>
+	<summary>Open-source electronics prototyping platform</summary>
+    <developer_name>Arduino LLC</developer_name>
```
```suggestion
    <developer_name>Arduino SA</developer_name>
```

joshuag1000 commented:
bot, build cc.arduino.IDE2

joshuag1000 commented:
bot, build cc.arduino.IDE2

hfiguiere commented on cc.arduino.IDE2.json:
```diff
@@ -0,0 +1,117 @@
+        "--device=all",
+        "--filesystem=home",
+        "--filesystem=xdg-run/keyring",
+        "--system-talk-name=org.gtk.vfs.Daemon",
```
mmm.

gvfs is on the session bus and in that case we just add `--talk-name=org.gtk.vfs.*`.

joshuag1000 replied:
ok i will update it to that.

thanks for the help and patients with this.

joshuag1000 replied:
however i am going to leave it on the system bus because otherwise the application complains about not being able to use the system bus.
`[3:0527/193631.191436:ERROR:bus.cc(392)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory`

barthalion commented:
/merge
